utils/middlewares/__tests__/hsl-jwt-validation.test.ts

```diff
@@ -0,0 +1,40 @@
+import * as E from "fp-ts/Either";
+import { hslJwtValidation } from "../hsl-jwt-validation-middleware";
+import { config } from "../../../__mocks__/config.mock";
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+describe(`Hub Spid Login JWT Validation Middleware`, () => {
+  it(`Should validate JWT and call hub-spid-login for introspection call,
+    - Should return JWT payload 
+    - Success -`, async () => {
+    const token = config.HUB_SPID_LOGIN_MOCK_TOKEN;
+
+    const jwtValidation = hslJwtValidation(token, config);
+    const result = await jwtValidation(token)();
+
+    expect(E.isRight(result)).toBe(true);
+
+    const resultPayload = E.fold(
+      () => null,
+      payload => payload
+    )(result);
+
+    expect(resultPayload).toMatchObject({
+      name: expect.anything() as string,
+      family_name: expect.anything() as string,
+      fiscal_number: expect.anything() as FiscalCode
+    });
+  });
+});

```

Comment:
I suggest to enrich this test. also the JWT provided in the config mock expires the 10th of August. another comment will address this problem
```suggestion
describe(`Hub Spid Login JWT Validation Middleware`, () => {
  it(`Should validate JWT and call hub-spid-login for introspection call,
    - Should return JWT payload 
    - Success -`, async () => {
    const mockIntrospectionCall = jest
      .spyOn(hslUtils, "introspectionCall")
      .mockReturnValueOnce(TE.right(ResponseSuccessJson({ active: true })));
    const token = config.HUB_SPID_LOGIN_MOCK_TOKEN;

    const jwtValidation = hslJwtValidation(token, config);
    const result = await jwtValidation(token)();

    expect(E.isRight(result)).toBe(true);
    expect(mockIntrospectionCall).toHaveBeenCalledTimes(1);
    expect(mockIntrospectionCall).toHaveBeenCalledWith(token, config);

    const resultPayload = E.fold(
      () => null,
      payload => payload
    )(result);

    expect(resultPayload).toMatchObject({
      name: "Carla",
      family_name: "Rossi",
      fiscal_number: "ISPXNB32R82Y766D"
    });
  });
});
```